platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/state/MerkleRoot.java

```diff
@@ -33,12 +33,23 @@ public interface MerkleRoot extends MerkleInternal {
     SwirldState getSwirldState();
 
     /**
-     * Get the platform state.
+     * Get readable platform state.
+     * Works on both - mutable and immutable {@link MerkleRoot} and, therefore, this method should be preferred.
      *
-     * @return the platform state
+     * @return immutable platform state
      */
     @NonNull
-    PlatformStateAccessor getPlatformState();
+    PlatformStateAccessor getReadablePlatformState();
```

Comment:
Just a thought (possibly a future PR?).
Would it make sense to split the platform state accessor interface?
`PlatformStateAccessor` might be the `get` methods, and `PlatformStateMutator extends PlatformStateAccessor` could then add the `set` methods.
That way we get more compiler support when code inadvertently tries to change a "readable" state.

Reply:
Sure, here it is https://github.com/hashgraph/hedera-services/issues/15413